cni.go

```diff
@@ -126,7 +127,7 @@ func (c *libcni) Remove(id string, path string, opts ...NamespaceOpts) error {
 		return err
 	}
 	for _, network := range c.networks {
-		if err := network.Remove(ns); err != nil {
+		if err := network.Remove(ns); err != nil && !strings.Contains(err.Error(), "no such file or directory") {
```

Comment:
It is not in spec, but it is the actual implementation. See containerd/cri#948.

Reply:
I was referring to this https://github.com/containernetworking/cni/pull/230/
In CRI we shd be passing an empty netns if it is removed on teardown ? and I beliveve its better for the cni plugins to handle it themselves instead of we silently ignoring the error. Can we only do this for loopback and host-device ? Just so we know what we are targeting.

Reply:
Yeah, we can do that. :)

Reply:
Hm, wait, there can be any arbitrary 3rd party plugins, are we sure that they all handle the empty network namespace case?

Reply:
Both the cni and kubenet in kubernetes have the same error handling. I know this is not ideal, and this is problem in CNI spec, because they don't define it clearly...

However, we need to live with it until it is fixed. :(

Reply:
> When CNI_NETNS and/or prevResult are not provided, the plugin should clean up as many resources as possible (e.g. releasing IPAM allocations) and return a successful response.

https://github.com/containernetworking/cni/commit/23a5d8e617027c87a84605426c3b228931c69826, this will be defined in CNI 0.7.0.